city_temperature/crud.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Optional
+
+from sqlalchemy.orm import Session
+
+from city_temperature import models
+from city_temperature.schemas import (
+    CityCreate,
+    CityUpdate,
+    TemperatureCreate,
+)
+
+
+def get_cities(db: Session) -> List[models.City]:
+    return db.query(models.City).all()
+
+
+def get_city(
+        db: Session,
+        city_id: int
+) -> Optional[models.City]:
+    return (
+        db.query(models.City).filter(
+            models.City.id == city_id
+        ).first()
+    )
+
+
+def get_city_by_name(db: Session, name: str) -> models.City:
```

Comment:
The function `get_city_by_name` should return `Optional[models.City]` instead of `models.City` to account for cases where a city with the given name does not exist.